src/main/java/eu/adlogix/appnexus/oas/client/service/AbstractOasService.java

```diff
@@ -0,0 +1,156 @@
+package eu.adlogix.appnexus.oas.client.service;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+import eu.adlogix.appnexus.oas.client.certificate.CertificateManager;
+import eu.adlogix.appnexus.oas.client.domain.PushLevel;
+import eu.adlogix.appnexus.oas.client.util.Credentials;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser;
+import eu.adlogix.appnexus.oas.client.xml.XmlRequestGenerator;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser.ResponseElementHandler;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser.ResponseObjectHandler;
+import eu.adlogix.appnexus.oas.utils.log.LogUtils;
+
+
+public class AbstractOasService {
```

Comment:
It's call Abstract, but it's not an abstract class? I would also introduce a OasService Interface that this class implementats